File: modules/agent/src/agent/mocks/repositories/customer_repo.py
import json
from pathlib import Path

DATA_PATH = Path(__file__).parent.parent / "data"

_customers: dict[str, dict] = {}


def _load_data():
    global _customers
    if _customers:
        return
    with open(DATA_PATH / "customers.json") as f:
        _customers = {c["id"]: c for c in json.load(f)["customers"]}


def get_all_customers() -> list[dict]:
    _load_data()
    return list(_customers.values())


def find_customer_by_id(customer_id: str) -> dict | None:
    _load_data()
    return _customers.get(customer_id)

File: modules/agent/src/mocks/repositories/customerRepo.ts
import { readFileSync } from 'fs';
import { dirname, join } from 'path';
import { fileURLToPath } from 'url';

const __dirname = dirname(fileURLToPath(import.meta.url));
const DATA_PATH = join(__dirname, '..', 'data');

export interface Customer {
  id: string;
  name: string;
  email: string;
  shipping_address: string;
}

let customers: Map<string, Customer> | null = null;

const loadData = (): void => {
  if (customers) return;

  const data = JSON.parse(readFileSync(join(DATA_PATH, 'customers.json'), 'utf-8'));
  customers = new Map(data.customers.map((c: Customer) => [c.id, c]));
};

export const getAllCustomers = (): Customer[] => {
  loadData();
  return Array.from(customers!.values());
};

export const findCustomerById = (customerId: string): Customer | undefined => {
  loadData();
  return customers!.get(customerId);
};
